libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/AbstractSpan.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+/**
+ * Base span
+ */
+public abstract class AbstractSpan implements Span {
+
+    /**
+     * name of the span
+     */
+    protected final String spanName;
```

Comment:
The fields should be `final`, right?

Reply:
I think you mean `private` as the fields are already `final`. Makes sense, made changes in new commit

Reply:
Sorry, yes, `private` it is